go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -220,8 +220,32 @@ func DemystifyPending(status v1.PodStatus) (pluginsCore.PhaseInfo, error) {
 								// PodInitializing -> Init containers are running
 								return pluginsCore.PhaseInfoInitializing(c.LastTransitionTime.Time, pluginsCore.DefaultPhaseVersion, fmt.Sprintf("[%s]: %s", finalReason, finalMessage), &pluginsCore.TaskInfo{OccurredAt: &c.LastTransitionTime.Time}), nil
 
-							case "CreateContainerConfigError", "CreateContainerError":
-								// This happens if for instance the command to the container is incorrect, ie doesn't run
+							case "CreateContainerError":
+								// This may consist of:
+								// 1. Transient errors: e.g. failed to reserve
+								// container name, container name [...] already in use
+								// by container
+								// 2. Permanent errors: e.g. no command specified
+								// To handle both types of errors gracefully without
+								// arbitrary pattern matching in the message, we simply
+								// allow a grace period for kubelet to resolve
+								// transient issues with the container runtime. If the
+								// error persists beyond this time, the corresponding
+								// task is marked as failed.
+								t := c.LastTransitionTime.Time
+								if v12.Now().After(t.Add(config.GetK8sPluginConfig().CreateContainerErrorGracePeriod.Duration)) {
```

Comment:
Yes, that’s a good point, but unfortunately lasttransition remains constant, and there doesn’t appear to be another time stamp we can use for this purpose, is there? Maybe leveraging events somehow?

Reply:
One option is to potentially leverage the kubelet's last heartbeat time. This might give us a conservative approximation of the amount of time a pod has spent stuck in `CreateContainerError`. The other alternative might be to get rid of grace period completely and just treat it like a system retryable error (grey area). Let it retry up to the configurable number of times (`max-node-retries-system-failures`), and fail after.

Reply:
For context, `kubectl` tracks age in a similar way: https://github.com/kubernetes/kubectl/blob/764d523af876fae3eeaf9dc4e44816c79f3e7f47/pkg/describe/describe.go#L5348, https://github.com/kubernetes/kubectl/blob/0f88fc6b598b7e883a391a477215afb080ec7733/pkg/cmd/top/top_pod.go#L277.